### ISSUE TITLE
Feature request: Clean react-query implementation 

### ISSUE THREAD

**Issue by botika** [anymaniax/orval#491]
I do something more clean in my implementation https://github.com/botika/forval 

Thanks!

**Comment by anymaniax** [comment]
Hello @botika, I will take a look thanks

**Comment by melloware** [comment]
@botika can you summarize what you do differently for React Query?  I am interested to hear what improvements you have made?

**Comment by melloware** [comment]
@botika going to close this for now as no response from you in a year.  Happy to reopen if you have suggestions or PR to offer.
